src/app/common/mongoose/paging-search.plugin.js

```diff
@@ -0,0 +1,86 @@
+const
+	deps = require('../../../dependencies'),
+	config = deps.config;
+
+function escapeRegex(str) {
+	return (`${str}`).replace(/[.?*+^$[\]\\(){}|-]/g, '\\$&');
+}
+
+function generateFilter(query) {
+	return Object.assign({}, query);
+}
+
+function generateSort(sortArr) {
+	return (sortArr || []).reduce((sort, element) => {
+		sort[element.property] = (element.direction === 'ASC') ? 1 : -1;
+		return sort;
+	}, {});
+}
+
+const pagingQuery = async (schema, filter, projection, options, sort, limit, offset, runCount = true, populate = []) => {
+	// Build the queries
+	const countQuery = schema.find(filter);
+	const resultsQuery = schema.find(filter, projection, options).sort(sort).skip(offset).limit(limit).maxTimeMS(config.maxTimeMS).populate(populate);
+
+	const countPromise = runCount ? countQuery.countDocuments().exec() : Promise.resolve(Number.MAX_SAFE_INTEGER);
+	const resultsPromise = resultsQuery.exec();
+
+	const [count, results] = await Promise.all([countPromise, resultsPromise]);
+
+	return { results, count };
+};
+
+// Generic contains regex search
+const searchContainsQuery = (schema, query, fields, search, limit, offset, sortArr, runCount) => {
+	const filter = generateFilter(query);
+	const sort = generateSort(sortArr);
+	const projection = {};
+	const options = {};
+
+	// Add search to the filter
+	if (null != search && '' !== search) {
+		if (null != fields && fields.length > 0) {
+			filter.$or = fields.map((element) => {
```

Comment:
Any chance you can add a parameter to the contains and searchtext query methods to allow us to switch this to an `$and` instead of an `$or`? We have a few screens where we use anded filters.

Reply:
If you want to use an `$and`, I think the proper method is to use the `query` parameter.  For example, on the admin user page, `query` is used for the role quick filter and `search` is used for the text search field.

Reply:
Ah-- Ok I think I see. Maybe we just need better documentation. lol